Sources/Modules/FilesModule.swift

```diff
@@ -1188,5 +1188,77 @@ public class FilesModule {
         }
     }
 
+    /// Creates a zip of multiple files and folders.
+    ///
+    /// - Parameters:
+    ///   - name: The name of the zip file to be created
+    ///   - items: Array of files or folders to be part of the created zip
+    ///   - completion: Returns a standard ZipDownload object or an error
+    private func createZip(
+        name: String,
+        items: [ZipDownloadItem],
+        completion: @escaping Callback<ZipDownload>
+    ) {
+        var body: [String: Any] = [:]
+        body["download_file_name"] = name
+        body["items"] = items.map { $0.bodyDictWithDefaultKeys }
+
+        boxClient.post(
+            url: URL.boxAPIEndpoint("/2.0/zip_downloads", configuration: boxClient.configuration),
+            json: body,
+            completion: ResponseHandler.default(wrapping: completion)
+        )
+    }
+
+    /// Creates a zip of multiple files and folders and downloads it.
+    ///
+    /// - Parameters:
+    ///   - name: The name of the zip file to be created
+    ///   - items: Array of files or folders to be part of the created zip
+    ///   - destinationURL: A URL for the location on device that we want to store the file once it has been downloaded
+    ///   - completion: Returns an empty response or an error
+    public func downloadZip(name: String, items: [ZipDownloadItem], destinationURL: URL, completion: @escaping Callback<ZipDownloadStatus>) {
+        createZip(name: name, items: items) { [weak self] result in
+            guard let self = self else {
+                completion(.failure(BoxSDKError(message: .instanceDeallocated("Unable to create Zip - FilesModule deallocated"))))
+                return
+            }
+
+            switch result {
+            case let .success(zip):
+                self.boxClient.download(
+                    url: zip.downloadUrl,
+                    downloadDestinationURL: destinationURL
+                ) { [weak self] zipDownloadResult in
+                    guard let self = self else {
+                        completion(.failure(BoxSDKError(message: .instanceDeallocated("Unable to download Zip - FilesModule deallocated"))))
+                        return
+                    }
+
+                    switch zipDownloadResult {
+                    case .success:
+                        self.getZipDownloadStatus(statusURL: zip.statusUrl, completion: completion)
```

Comment:
We need to combine this with zip.nameConflicts, returned from the call to createZip, so that callers to downloadZip get back both the download status AND the name conflicts (since we made createZip private).  You can follow [the example in .Net](https://github.com/box/box-windows-sdk-v2/pull/700/files#diff-7fb373d03b85bf95bfac5c4d09c2ea5816fc6a69ef62bb97191733815a70881eR1360).